libraries/AC_AttitudeControl/AC_AttitudeControl_Logging.cpp

```diff
@@ -0,0 +1,82 @@
+#include <AP_Logger/AP_Logger_config.h>
+
+#if HAL_LOGGING_ENABLED
+
+#include "AC_AttitudeControl.h"
+#include "AC_PosControl.h"
+#include <AP_Logger/AP_Logger.h>
+#include <AP_Scheduler/AP_Scheduler.h>
+#include "LogStructure.h"
+
+// Write an ANG packet
+void AC_AttitudeControl::Write_ANG() const
+{
+    Vector3f targets = get_att_target_euler_rad() * RAD_TO_DEG;
+    // Retrieve quaternion body attitude
+    Quaternion attitude_body;
+    _ahrs.get_quat_body_to_ned(attitude_body);
+    // calculate the attitude body euler angles
+    Vector3f attitude_body_angles;
+    attitude_body.to_euler(attitude_body_angles);
+    attitude_body_angles *= RAD_TO_DEG;
```

Comment:
Why not use `degrees(_ahrs.roll)`, `degrees(_ahrs.pitch)`, `degrees(_ahrs.yaw)`? AHRS has done the conversion for us already.

Reply:
Leonard wants this as separate from AHRS as possible. Other changes incoming in future PR's

Reply:
This is attitude referencing used by the Attitude Controller, independent of its source.

Reply:
For now you are right. We should just use _ahrs.ect versions.

Reply:
Fixed